src/components/tl/cuda/reduce_scatterv/reduce_scatterv_linear.c

```diff
@@ -0,0 +1,443 @@
+/**
+ * Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "reduce_scatterv.h"
+#include "components/ec/ucc_ec.h"
+#include "tl_cuda_cache.h"
+#include "utils/arch/cpu.h"
+#include "utils/arch/cuda_def.h"
+
+/*
+ * fragmented buffered copy linear reduce scatterv algorithm
+ *
+ * Description:
+ *  Definitions:
+ *      blockI  - full send buffer at Rank I
+ *      fragI_J - fragment of send buffer at Rank I and step J
+ *      NS      - number of steps
+ *      NF      - number of fragments
+ *      N       - team size
+ *
+ *  Setup:
+ *      max_frag_size = ucc_min(ucc_max(block1, block2, ..., block N),
+ *                              scratch_size / 2 / N)
+ *      NF            = ucc_max(block1, block2, ..., block N) / max_frag_size
+ *      NS            = 1 + NF
+ *
+ *  Algorithm
+ *      for rank R
+ *      step 1:    copy fragR_1 to remote scratch buffers for all ranks
```

Comment:
I think based on the code below, this is step 0? Would help avoiding having two different 'step 1' in the description of the algorithm